submissions/stazrouti/solution.js

```diff
@@ -0,0 +1,70 @@
+//to run the code use node solution.js in the terminal in this directory
+// Importing the fs module for file operations
+const fs = require('fs');
+
+// Reading the input.txt file
+fs.readFile('../../input.txt', 'utf8', (err, data) => {
```

Comment:
This loads the whole file in memory I believe. It can trigger ab OOM

Reply:
but the memory usage is not indicated in the requirement !!

![angry](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExajRyN2cxbXljaWttYjdjemdienc0bnFxcXk5ZnBmeHBuYWt5dzRwZCZlcD12MV9naWZzX3NlYXJjaCZjdD1n/3o9bJX4O9ShW1L32eY/giphy.gif)

Reply:
@aboullaite 
but there is no need for a lot of ressources

Reply:
Loading the whole file in memory would require 22G, plus the overhead of any data structure to handle your code, and then everything else. That's a lot of memory

Reply:
sorry but no need for 22 G, i try it